### ISSUE TITLE
Click Listeners not invoked when Slot is empty

### ISSUE THREAD

**Issue by GitGraf** [MinusKube/SmartInvs#79]
When I add a click listener via the `Builder#listener()` method, the consumer is not invoked if the clicked slot and the cursor is empty.

Example:
```java
SmartInventory inv = SmartInventory.builder()
    .manager(myManager)
    .provider(new MyInvProvider())
    .listener(new InventoryListener<InventoryClickEvent>(InventoryClickEvent.class, e -> {
        e.getWhoClicked().sendMessage("Slot: " + e.getSlot());
    }))
    .build();

// Provider
class MyInvProvider implements InventoryProvider {
    @Override
    public void init(Player player, InventoryContents contents) {
        contents.fillRow(1, ClickableItem.empty(new ItemStack(Material.DIRT)));
    }
    @Override
    public void update(Player player, InventoryContents contents) {}
}
```
The clicked slot is only displayed when clicking on one of the dirt blocks in the inventory.

**Comment by Oscaz** [comment]
This caused by the check for InventoryAction.NOTHING in InventoryManager,

```java
if (e.getAction() == InventoryAction.NOTHING && e.getClick() != ClickType.MIDDLE) {
    e.setCancelled(true);
    return;
}
```

**Comment by MinusKube** [comment]
Fixed by #186
